modules/afmongodb/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+find_package(libmongoc-1.0)
+
+module_switch(ENABLE_MONGODB "Enable mongodb destination driver" libmongoc-1.0_FOUND)
+
+if (libmongoc-1.0_FOUND)
+
+  set(AFMONGODB_HEADERS
+    "afmongodb.h"
+    "afmongodb-parser.h"
+    "afmongodb-private.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.h"
+  )
+
+  set(AFMONGODB_SOURCES
+    "afmongodb.c"
+    "afmongodb-parser.c"
+    "${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.c"
+  )
+
+set(AFMONGODB_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+generate_y_from_ym(modules/afmongodb/afmongodb-grammar)
+bison_target(AFMongoDBGrammar
+  ${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.y
+  ${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.c
+  COMPILE_FLAGS ${BISON_FLAGS})
+
+include_directories (afmongodb SYSTEM PRIVATE ${MONGOC_INCLUDE_DIRS})
+add_library(afmongodb SHARED ${AFMONGODB_SOURCES})
+target_include_directories (afmongodb PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
okay, then why we added `CMAKE_CURRENT_SOURCE_DIR` to `target_include_directories()` in other places?

I'll fix only the `include_directories()`, other changes are welcomed in a patch, or, in a separate PR... that normalizes all the modules' CMake files ;)

Reply:
> okay, then why we added CMAKE_CURRENT_SOURCE_DIR to target_include_directories() in other places?

I added them as PUBLIC interface to every single module in 2018:
https://github.com/syslog-ng/syslog-ng/pull/2007/commits/40ed28550117242ed30a5afd09073504b8df3fe2

> I'll fix only the include_directories(), other changes are welcomed in a patch, or, in a separate PR... that normalizes all the modules' CMake files ;)

That patch above was the normalization of all CMake modules. It was more than a year ago, I guess things have change since then. I can do it again.

Reply:
I see.
There are quite a few modules where this pattern is not followed (afsmtp,afstomp,appmodel,basicfuncs,confgen,cryptofuncs,geoip2, and so on) - which is not good.